storage/src/persistent/database.rs

```diff
@@ -82,61 +133,72 @@ pub trait KeyValueStoreWithSchema<S: KeyValueSchema> {
     /// * `key` - Value of key specified by schema
     fn get(&self, key: &S::Key) -> Result<Option<S::Value>, DBError>;
 
-    /// Read all entries in database.
-    ///
-    /// # Arguments
-    /// * `mode` - Reading mode, specified by RocksDB, From start to end, from end to start, or from
-    /// arbitrary position to end.
-    fn iterator(&self, mode: IteratorMode<S>) -> Result<IteratorWithSchema<S>, DBError>;
-
-    /// Starting from given key, read all entries to the end.
-    ///
-    /// # Arguments
-    /// * `key` - Key (specified by schema), from which to start reading entries
-    fn prefix_iterator(&self, key: &S::Key) -> Result<IteratorWithSchema<S>, DBError>;
-
     /// Check, if database contains given key
     ///
     /// # Arguments
     /// * `key` - Key (specified by schema), to be checked for existence
     fn contains(&self, key: &S::Key) -> Result<bool, DBError>;
 
-    /// Insert new key value pair into WriteBatch.
+    /// Check, if database contains given key
     ///
     /// # Arguments
-    /// * `key` - Value of key specified by schema
-    /// * `value` - Value to be inserted associated with given key, specified by schema
-    fn put_batch(
-        &self,
-        batch: &mut WriteBatch,
-        key: &S::Key,
-        value: &S::Value,
-    ) -> Result<(), DBError>;
+    /// * `key` - Key (specified by schema), to be checked for existence
+    fn retain(&self, predicate: &dyn Fn(&S::Key) -> bool) -> Result<(), DBError>;
```

Comment:
I am confused by this name, it is called `retain` but the description says it just checks if a given key is there.

Reply:
Ohhhh, it is the same comment as `contains` above. This one needs to be updated.